src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -405,25 +729,11 @@ private BoundExpression ConvertConcatExprToString(SyntaxNode syntax, BoundExpres
 
             // Evaluate toString at the last possible moment, to avoid spurious diagnostics if it's missing.
             // All code paths below here use it.
-            var objectToStringMethod = UnsafeGetSpecialTypeMethod(syntax, SpecialMember.System_Object__ToString);
+            var objectToStringMethod = UnsafeGetSpecialTypeMethod(expr.Syntax, SpecialMember.System_Object__ToString);
```

Comment:
>expr.Syntax [](http://example.com/codeflow?start=66&length=11)

I am not sure why is this changed. Is there an effect that can be observed through added tests? If not, I suggest reverting this change and the like. I can see that this function, perhaps, can benefit from a clean-up in terms of what syntax nodes it is using, and, quite possibly, we can get rid of the ```syntax``` parameter completely. However, I would prefer if we stay focused in this PR and avoid making unnecessary changes.

Reply:
This changes error location about missing `object.ToString()` from the whole expression to the argument that requested it. That just makes more sense to report it there. The commit I changed it in was about testing missing `object.ToString()` for span-based concat and probably all tests, added in that commit, verify this new error location. As far as I remember, only 1 existing test was modified to account for this change.

Reply:
> This changes error location about missing object.ToString() from the whole expression to the argument that requested it. That just makes more sense to report it there

In my opinion, this is not related to the goal of this PR. If you feel like this location is worth fixing, feel free to submit a separate PR. Explaining what you are doing and why.

Reply:
See https://github.com/dotnet/roslyn/pull/71890